### DISCUSSION TITLE
Error when selecting Input Field

### DISCUSSION THREAD

**Discussion by JaySecurity** [jeffreylanters/react-unity-webgl#319] [Help Wanted!]
Unity Version 2021.2.7 using UIToolkit 

The webgl build works fine on its own, when brought into react clicking on the input field causes TypeError Cannot set properties of null at _FocusExternalInput. Is UIToolkit supported?  or is there something else I may be missing? 
 
![gamelogin](https://user-images.githubusercontent.com/33810491/152376705-d850b7bc-9323-42a7-bca3-e4e43e119a2c.png)
t

**Comment by jeffreylanters** [comment]
Everything that works in a stand alone build should work in this module. Could you provide some more details?

**Reply by DivRBL** [reply]
> Everything that works in a stand alone build should work in this module. Could you provide some more details?

The ui in the build is built using the unity ui toolkit which appears to send messages to the browser console by default when the input field is selected..Have a feeling it may have something to do with that. The rest of the buttons implemented with uitoolkit etc are completely intractable though

**Reply by jeffreylanters** [reply]
I've never used UIToolKit before, but it might alters the build's HTML, which doesn't get used in React. It seems like a keyboard capturing issue. Could you check this section in documentation and see if this helps with your problem?

https://github.com/jeffreylanters/react-unity-webgl#tab-index-and-input-keyboard-capturing

**Reply by DivRBL** [reply]
> I've never used UIToolKit before, but it might alters the build's HTML, which doesn't get used in React. It seems like a keyboard capturing issue. Could you check this section in documentation and see if this helps with your problem?
> 
> https://github.com/jeffreylanters/react-unity-webgl#tab-index-and-input-keyboard-capturing

These are methods defined in our jslib that the webgl integration is using. Is there anything that needs to be done on the react side to account for the unity webgl jsIntegration methods?
